moped-editor/src/views/projects/projectView/SignalComponentAutocomplete.js

```diff
@@ -16,12 +16,20 @@ import { filterOptions } from "src/utils/autocompleteHelpers";
  * @param {Function} onChange - callback function to run when the signal is changed for React Hook Form
  * @param {Object} value - the signal feature to set as the value of the autocomplete from React Hook Form
  * @param {Function} onOptionsLoaded - callback function to run when the options are loaded
+ * @param {String} signalType - either PHB or TRAFFIC
  * @return {JSX.Element}
  */
 const SignalComponentAutocomplete = React.forwardRef(
-  ({ classes, onChange, value, onOptionsLoaded }, ref) => {
+  ({ classes, onChange, value, onOptionsLoaded, signalType }, ref) => {
     const { features, loading, error } = useSocrataGeojson(SOCRATA_ENDPOINT);
 
+    // Filter returned results to the signal type chosen - PHB or TRAFFIC
+    const featuresFilteredByType = features?.filter(
+      (feature) =>
+        feature.properties.signal_type.toLowerCase() ===
+        signalType.toLowerCase()
+    );
```

Comment:
it's not a long list, but since we're re-rendering on any component form input change, it's probably best to memoize this?

Reply:
This is the right call. I updated this, and it also plays nicely with the fix for the issue that Patrick caught in his review below. 🙏 🚀